lib/chef/resource/cpan_module.rb

```diff
@@ -0,0 +1,143 @@
+#
+# Copyright:: Copyright 2009-2018, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/resource"
+
+class Chef
+  class Resource
+    class CpanModule < Chef::Resource
+      resource_name :cpan_module
+      provides :cpan_module
+
+      description "Use the cpan_module resource to install or uninstall perl CPAN modules."
+      introduced "14.0"
+
+      property :module_name,
+               String,
+               description: "The name of the module if it's different than the name of the resource.",
+               name_property: true
+
+      property :force,
+               [TrueClass, FalseClass],
+               description: "To force the install within cpanm.",
+               default: false
+
+      property :test,
+               [TrueClass, FalseClass],
+               description: "To do a test install.",
+               default: false
+
+      property :version,
+               String,
+               description: "Any version string cpanm would find acceptable."
+
+      property :cwd,
+               String,
+               description: "A path to change into before running cpanm."
+
+      property :cpanm_binary,
+               String,
+               description: "The path of the cpanm binary.",
+               default: "cpanm"
+
+      action :install do
+        description "Install the module."
+
+        declare_resource(:execute, "CPAN :install #{new_resource.module_name}") do
+          cwd current_working_dir
+          command cpanm_install_cmd
+          environment "HOME" => current_working_dir, "PATH" => "/usr/local/bin:/usr/bin:/bin"
```

Comment:
this should be in `({})`'s.  also its missing in the uninstall action.  also it is some very interesting hardcoding that seems a bit brittle.